### PR TITLE
Making some turrets to spawn with less ammo

### LINKED ISSUE #53897: Use spawn_data to have some turrets start with less ammo
### Is your feature request related to a problem? Please describe.

Turret ammo spawns are an unfortunate balance issue: turrets are too easy a loot pinata for many players.

### Solution you would like.

Ideally we should continue to reduce the use of turrets overall until eventually most of them are phased out except in particular special cases. However, a more short-term solution and one that we'll need anyway is to check where a turret is spawning and manually review `spawn_data` to ensure that the ammunition levels are appropriate to spawn location.

- Turrets that are spawning in places where no conflict occurred should have lots of ammo
- Turrets spawning in active combat areas should have very little ammo, having spent days burning it on zombies that just keep getting back up.
- Turrets spawning in ambiguous locations where there may have been combat, it's unsure, should have a very random amount of ammo.

Spawn_data is applied at the mapgen level, for example:
```
        {
          "monster": "mon_crows_m240",
          "x": 10,
          "y": 1,
          "spawn_data": { "ammo": [ { "ammo_id": "762_51", "qty": [ 1200, 1600 ] } ] }
        }
```

### Describe alternatives you have considered.

Having far fewer turrets and replacing most of them with lore-appropriate enemies fulfilling the same function.

### Additional context

Ammunition still should be fairly plentiful, but it should mostly be a thing you are looting, not blowing up AI turrets for. The current mechanism leads to a very counterintuitive way to get a lot of just a few specific ammo types.

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Balance "tweaking some turret's ammo"

#### Purpose of change

Closes #53897

#### Describe the solution

The lab related turrets has 30-90 bullets inside (because `The ones in labs may also suit having their total ammo reduced since they're not necessarily complete milspec` - Erk)
Missile silo above-ground turrets has 200-800 bullets (because they are outside, and probably kill a bunch of zeds)
GROUP_TURRET has random amount of ammo up to their `starting_ammo` (it is not used yet, so i just left it at random)

Numbers were totally random, so if you have better idea i will change them

#### Testing
Teleported to some labs, spawned some silos, killed a bunch of turrets, got the right amount of ammo
image
image

### CODE REVIEW

**Comment by taatu** [comment]
This type of change has a history of PRs that get denied: #50413 #49145

Since #55790, the only aboveground turrets should be those behind locked metal doors, so the ammo counts should probably match underground turrets

File: data/json/mapgen/gas_stations/s_gas_b11.json [hunk 1]
```diff
@@ -75,7 +75,10 @@
         ],
         "n": { "item": "SUS_bathroom_sink", "chance": 90 }
       },
-      "monster": { "q": { "monster": "mon_turret_rifle" }, "r": { "monster": "mon_zombie_soldier" } }
+      "monster": {
+        "q": { "monster": "mon_turret_rifle", "spawn_data": { "ammo": [ { "ammo_id": "556", "qty": [ 30, 90 ] } ] } },
```
Comment (LyleSY) [inline data/json/mapgen/gas_stations/s_gas_b11.json:79]: Why is there a turret at this gas station anyway? Is this an error?

Comment (LyleSY) [reply data/json/mapgen/gas_stations/s_gas_b11.json:79]: Ah, I see. This is a secret underground bunker thing. Seems fine.

**Review by LyleSY — APPROVED** [review]

**Comment by LyleSY** [comment]
I like that this addresses the missile silo issue, though I'm not sure it's better than just replacing the turrets there with a bunch of military zombies

**Pull request merged**